security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc

```diff
@@ -0,0 +1,55 @@
+= <HTTP Basic Authentication - Silent Mode>
+:author:            Diana Vilkolakova
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          basic,elytron,silent
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Using HTTP Basic authentication with Undertow it was possible to set silent mode. Basic authentication in silent mode will send a challenge only if the request contained authorization header, otherwise it is assumed another method will send the challenge. This behaviour will allow to combine basic auth with form auth, so human users will use form based auth and programmatic clients can use basic authentication to log in. This RFE is to support the same behaviour in Elytron.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11479[WFLY-11479]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1154[EAP7-1154]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[dvilkola]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+Wildfly security
+
+Elytron
+
+=== Other Interested Projects
+
+Undertow Core - Basic authentication in Undertow has the same mode available
+
+== Requirements
+
+=== Hard Requirements
+
+This will involve processing optional property `silent` set along with basic authentication. If the property is set as true and request does not contain Authorization header, response will not include `WWW-Authenticate` header and http status code will not be `401`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
```

Comment:
Do you plan to add some tests?

Reply:
if tests are necessary, I will add them. The code is very similar to implementation in Undertow here: https://github.com/undertow-io/undertow/blob/master/core/src/main/java/io/undertow/security/impl/BasicAuthenticationMechanism.java

Reply:
We do have some HTTP auth tests with Elytron, I would expect this should be simple to add there.

Reply:
@darranl thank you, I'll add tests and will update analysis doc